Metric to report time spent fetching and analyzing segments (#14752)
apache/druid: docs/operations/metrics.md

|`ingest/kinesis/avgLag/time`|Average lag time in milliseconds between the current message sequence number consumed by the Kinesis indexing tasks and latest sequence number in Kinesis across all shards. Minimum emission period for this metric is a minute.|`dataSource`, `stream`, `tags`|Greater than 0, up to max Kinesis retention period in milliseconds. |
|`ingest/kinesis/partitionLag/time`|Partition-wise lag time in milliseconds between the current message sequence number consumed by the Kinesis indexing tasks and latest sequence number in Kinesis. Minimum emission period for this metric is a minute.|`dataSource`, `stream`, `partition`, `tags`|Greater than 0, up to max Kinesis retention period in milliseconds. |

### Compaction metrics

> These metrics are only reported by [compaction tasks](../data-management/compaction.md).

Reviewer: ```suggestion
[Compaction tasks](../data-management/compaction.md) emit the following metrics.
```

> |Metric|Description|Dimensions| Normal value                                                                                           |

Reviewer: ```suggestion
|Metric|Description|Dimensions|Normal value|
```

> |------|-----------|----------|--------------------------------------------------------------------------------------------------------|

Reviewer: ```suggestion
|Metric|Description|Dimensions|Normal value|
|------|-----------|------------|-------------|
```

> |`compact/segmentAnalyzer/fetchAndProcessMillis`|Time taken to fetch and process segments to infer the schema for the compaction task to run.|`dataSource`, `taskId`, `taskType`, `groupId`,`tags`| Varies. A high value indicates compaction tasks will speed up from explicitly setting the data schema. |

Reviewer: ```suggestion
|`compact/segmentAnalyzer/fetchAndProcessMillis`|The time it takes to fetch and process segments to infer the schema for the compaction task to run.|`dataSource`, `taskId`, `taskType`, `groupId`,`tags`| Varies. A high value indicates compaction tasks will speed up from explicitly setting the data schema. |
```
